src/modules/test/RecipeInvoker.cpp

```diff
@@ -80,13 +78,12 @@ void RecipeInvoker::TestBody()
             if (m_recipe.m_payload.size())
             {
                 JSON_Value *recipe_payload = json_parse_string(m_recipe.m_payload.c_str());
-                JSON_Value *returned_payload = json_parse_string(payload);
+                JSON_Value *returned_payload = json_parse_string(payloadString.c_str());
 
-                std::string recipe_payload_str(payload, payloadSize);
                 EXPECT_NE(nullptr, recipe_payload) << "Failed to parse recipe payload" << std::endl << "JSON: " << m_recipe.m_payload.c_str() << std::endl;
-                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << payloadString.c_str() << std::endl;
```

Comment:
Similar to above validation above, are we validating JSON objects against their respective MIM type?

Reply:
yes, but this happens before the JSON validation, this code is just responsible for the JSON comparison